autokeras/tuner.py

```diff
@@ -67,6 +71,27 @@ def load_trial(self, trial):
         path = os.path.join(trial.directory, filename)
         self.hypermodel.load_preprocessors(path)
 
+    @staticmethod
+    def add_earlystopping_callback(callbacks):
+        if not callbacks:
+            callbacks = []
+
+        try:
+            callbacks = copy.deepcopy(callbacks)
+        except:
+            raise ValueError(
+                'All callbacks used during a search '
+                'should be deep-copyable (since they are '
+                'reused across executions). '
+                'It is not possible to do `copy.deepcopy(%s)`' %
+                (callbacks,))
+
+        if not [callback for callback in callbacks
+                if isinstance(callback, tf.keras.callbacks.EarlyStopping)]:
+            callbacks.append(tf.keras.callbacks.EarlyStopping(patience=30))
```

Comment:
30 seems like a fairly arbitrary config. At least there should be a comment to explain this choice. Perhaps it should be made configurable by the user?

Reply:
I will add the comment. 30 is just based on my experience.

The user should pass their own earlystopping in the callbacks if they would like to configure it.